src/Microsoft.DotNet.Darc/src/DarcLib/LocalGitClient.cs

```diff
@@ -467,6 +454,70 @@ private static void CleanRepoAndSubmodules(LibGit2Sharp.Repository repo, ILogger
             }
         }
 
+        private static void SafeCheckout(LibGit2Sharp.Repository repo, string commit, LibGit2Sharp.CheckoutOptions options, ILogger log)
+        {
+            try
+            {
+                log.LogDebug($"Trying safe checkout of {repo.Info.WorkingDirectory} at {commit}");
+                LibGit2Sharp.Commands.Checkout(repo, commit, options);
+            }
+            catch (Exception e) when (e is LibGit2Sharp.InvalidSpecificationException || e is LibGit2Sharp.NameConflictException || e is LibGit2Sharp.LibGit2SharpException)
+            {
+                log.LogWarning($"Couldn't check out one or more files, possibly due to path length limitations ({e.ToString()}).  Attempting to checkout by individual files.");
```

Comment:
>LogWarning [](start = 20, length = 10)

Do we ever plan to use this flow in AzDo builds? If so, this warning might make the build to fail if `warnAsError` is set to true which is the case by default

Reply:
I'm currently using this in an AzDo build and it doesn't seem to be triggering it.  I'm not opposed to it doing so though - I think someone having to look at this and say it's okay that files aren't getting checked out is desirable behavior.

---
In reply to: [285644361](https://github.com/dotnet/arcade-services/pull/422#discussion_r285644361) [](ancestors = 285644361)

Reply:
I totally agree, just wanted to point that since using warnings has hit us in the past and I remember we ended up switching to a info/debug log.